src/ring/middleware/json.clj

```diff
@@ -64,14 +84,29 @@
   Use the standard Ring middleware, ring.middleware.keyword-params, to
   convert the parameters into keywords."
   {:arglists '([handler] [handler options])}
-  [handler & [{:keys [bigdecimals? malformed-response]
-               :or {malformed-response default-malformed-response}}]]
-  (fn [request]
-    (if-let [[valid? json] (read-json request {:bigdecimals? bigdecimals?})]
-      (if valid?
-        (handler (assoc-json-params request json))
-        malformed-response)
-      (handler request))))
+  [handler & [{:keys [malformed-response]
+               :or {malformed-response default-malformed-response}
+               :as options}]]
+  (fn
+    ([request]
+     (if-let [request (json-params-request request options)]
+       (handler request)
+       (handler malformed-response)))
+    ([request respond raise]
+     (if-let [request (json-params-request request options)]
+       (handler request respond raise)
+       (respond malformed-response)))))
+
+(defn- json-response-response
```

Comment:
This needs to be public, and maybe we should just call it `json-response`, as `json-response-response` is a little redundant in this case :)